lib/unleash/constraint.rb

```diff
@@ -55,9 +55,9 @@ def matches_context?(context)
 
     def self.on_valid_date(val1, val2)
       val1 = DateTime.parse(val1)
-      val2 = DateTime.parse(val2)
+      val2 = val2.is_a?(DateTime) ? val2 : DateTime.parse(val2)
       yield(val1, val2)
-    rescue ArgumentError
+    rescue ArgumentError, TypeError, Date::Error
```

Comment:
According to the [spec](https://github.com/ruby/ruby/blob/master/spec/ruby/library/datetime/parse_spec.rb), it appears that only ArgumentError is returned, but in reality, a different error occurs.

```rb
$ irb
irb(main):001> require 'date_core'
=> true
irb(main):002> DateTime.parse("1")
(irb):2:in `parse': invalid date (Date::Error)

DateTime.parse("1")
               ^^^
	from (irb):2:in `<main>'
	from /Users/marocchino/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/irb-1.13.2/exe/irb:9:in `<top (required)>'
	from /Users/marocchino/.local/share/rtx/installs/ruby/3.2.2/bin/irb:25:in `load'
	from /Users/marocchino/.local/share/rtx/installs/ruby/3.2.2/bin/irb:25:in `<main>'
irb(main):003> DateTime.parse(1)
(irb):3:in `parse': no implicit conversion of Integer into String (TypeError)

DateTime.parse(1)
               ^
	from (irb):3:in `<main>'
	from /Users/marocchino/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/irb-1.13.2/exe/irb:9:in `<top (required)>'
	from /Users/marocchino/.local/share/rtx/installs/ruby/3.2.2/bin/irb:25:in `load'
	from /Users/marocchino/.local/share/rtx/installs/ruby/3.2.2/bin/irb:25:in `<main>'
```

Reply:
That is a very good and valid point!